File: 백준/Bronze/1236. 성 지키기/성 지키기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st1 = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st1.nextToken());
        int m = Integer.parseInt(st1.nextToken());
        char[][] mtx = new char[n][m];
        for (int i = 0; i < n; i++) {
            char[] li = br.readLine().toCharArray();
            for (int j = 0; j < m; j++) {
                mtx[i] = li;
            }
        }
        int cnt1 = 0;
        int cnt2 = 0;

        for (int i = 0; i < n; i++) {
            boolean state = false;
            for (int j = 0; j < m; j++) {
                if (mtx[i][j] == 'X') {
                    state = true;
                    break;
                }
            }
            if (!state) cnt1++;
        }
        
        for (int i = 0; i < m; i++) {
            boolean state = false;
            for (int j = 0; j < n; j++) {
                if (mtx[j][i] == 'X') {
                    state = true;
                    break;
                }
            }
            if (!state) cnt2++;
        }
        
        System.out.println(Math.max(cnt1,cnt2));
    }
}

File: 백준/Bronze/1236. 성 지키기/성 지키기.py
import sys
input = sys.stdin.readline

n,m = map(int,input().split())
mtx = [list(input().rstrip('\n')) for _ in range(n)]
cnt1 = 0
cnt2 = 0
for i in range(n):
    state = False
    for j in range(m):
        if mtx[i][j] == "X":
            state = True
    if not state: cnt1 += 1

for i in range(m):
    state = False
    for j in range(n):
        if mtx[j][i] == "X":
            state = True
    if not state: cnt2 += 1
print(max(cnt1,cnt2))